docs(alerting): minor enhancements to intro concepts and example (#102023)
grafana/grafana: docs/sources/alerting/fundamentals/alert-rules/annotation-label.md

Annotations add additional information to alert instances, helping responders identify and address potential issues.

> Use annotations to make alerts self-explanatory, ensuring they provide enough information for first responders to investigate without needing deep knowledge of the alert setup.

Reviewer: ```suggestion
Create clear and self-explanatory annotations so that first responders can investigate without needing deeper knowledge of the alert setup.
```

Reviewer: just one minor suggestion to tighten up the sentence. 
otherwise: 👍.
